src/cli/make_plots.py

```diff
@@ -150,17 +179,24 @@ def estimate_masses(self) -> dict:
             for trial in range(num_trials):
                 init_trial_length = subject.getTrialLength(trial)
 
-                # Check that the trial has at least a dynamics pass.
-                # In future builds, we may need to modify this statement if more processing pass types added.
-                # Also, this checks for an initial idea of the trial length. True trial length comes from only valid frames.
-                if ((subject.getTrialNumProcessingPasses(trial) >= self.target_num_processing_passes)
-                        and (init_trial_length >= self.min_trial_length)):
-
-                    # # Extra check for Moore dataset: don't process first segment bc likely lacks GRFs in beginning
-                    # if "Moore" in subj_path:
-                    #     if subject.getTrialSplitIndex(trial) == 0:
-                    #         print(f"SKIPPING TRIAL {trial} BECAUSE SPLIT INDEX IS 0")
-                    #         continue
+                # Do a series of checks to see if okay to process this trial
+                has_dynamics = False
+                processing_passes = set()
+                for pass_ix in range(subject.getTrialNumProcessingPasses(trial)):
+                    processing_passes.add(subject.getProcessingPassType(pass_ix))
+                    if subject.getProcessingPassType(pass_ix) == nimble.biomechanics.ProcessingPassType.DYNAMICS:
+                        has_dynamics = True
```

Comment:
It's not a big deal, but you could add a `break` after `has_dynamics = True`, because there's no point continuing the loop after that - you already know you have dynamics.